### ISSUE TITLE
Field Arrays don't work

### ISSUE THREAD

**Issue by kselax** [erikras/redux-form#4456]
I have the code
```javascript
import React from 'react'
import ReactDOM from 'react-dom'
import { Provider, connect } from 'react-redux'
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import { reducer as formReducer, Field, reduxForm, FieldArray } from 'redux-form'

const reducer1 = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TODO': {
      const { id, name } = action.payload
      return [
        ...state,
        { id, name }
      ]
    }
    default: return state
  }
}

const rootReducer = combineReducers({
  reducer1,
  form: formReducer
})

const logger = createLogger({
  collapsed: true
})

const store = createStore(rootReducer, applyMiddleware(thunk, logger))

store.dispatch({
  type: 'ADD_TODO',
  payload: { id: 1, name: 'user1'}
})

store.dispatch({
  type: 'ADD_TODO',
  payload: { id: 2, name: 'user2'}
})

console.log('state = ', store.getState());

class App extends React.Component {
  submit = values => {
    console.log('values = ', values);
  }
  render() {
    const { reducer1 } = this.props
    return (
      <div>
        <h1>App</h1>
        <ul>
          {reducer1.map((e, i) => <li key={i}>
              {e.name}
            </li>)}
        </ul>
        <FieldArraysForm onSubmit={this.submit} />
      </div> 
    )
  }
}
App = connect(state => ({ ...state }))(App)

const validate = values => {
  console.log('values = ', values);
  const errors = {};
  if (!values.clubName) {
    errors.clubName = 'Required';
  }
  if (!values.members || !values.members.length) {
    errors.members = { _error: 'At least one member must be entered' };
    console.log('here we are');
  }
  // else {
  //   const membersArrayErrors = [];
  //   values.members.forEach((member, memberIndex) => {
  //     const memberErrors = {};
  //     if (!member || !member.firstName) {
  //       memberErrors.firstName = 'Required';
  //       membersArrayErrors[memberIndex] = memberErrors;
  //     }
  //     if (!member || !member.lastName) {
  //       memberErrors.lastName = 'Required';
  //       membersArrayErrors[memberIndex] = memberErrors;
  //     }
  //     if (member && member.hobbies && member.hobbies.length) {
  //       const hobbyArrayErrors = [];
  //       member.hobbies.forEach((hobby, hobbyIndex) => {
  //         if (!hobby || !hobby.length) {
  //           hobbyArrayErrors[hobbyIndex] = 'Required';
  //         }
  //       });
  //       if (hobbyArrayErrors.length) {
  //         memberErrors.hobbies = hobbyArrayErrors;
  //         membersArrayErrors[memberIndex] = memberErrors;
  //       }
  //       if (member.hobbies.length > 5) {
  //         if (!memberErrors.hobbies) {
  //           memberErrors.hobbies = [];
  //         }
  //         memberErrors.hobbies._error = 'No more than five hobbies allowed';
  //         membersArrayErrors[memberIndex] = memberErrors;
  //       }
  //     }
  //   });
  //   if (membersArrayErrors.length) {
  //     errors.members = membersArrayErrors;
  //   }
  // }
  return errors;
};

const renderField = ({ input, label, type, meta: { touched, error } }) => (
  <div>
    <label>{label}</label>
    <div>
      <input {...input} type={type} placeholder={label} />
      {touched && error && <span>{error}</span>}
    </div>
  </div>
);

const renderHobbies = ({ fields, meta: { error } }) => (
  <ul>
    <li>
      <button type="button" onClick={() => fields.push()}>
        Add Hobby
      </button>
    </li>
    {fields.map((hobby, index) => (
      <li key={index}>
        <button
          type="button"
          title="Remove Hobby"
          onClick={() => fields.remove(index)}
        />
        <Field
          name={hobby}
          type="text"
          component={renderField}
          label={`Hobby #${index + 1}`}
        />
      </li>
    ))}
    {error && <li className="error">{error}</li>}
  </ul>
);

const renderMembers = ({ fields, meta: { touched, error, submitFailed } }) => (
  <ul>
    <li>
      <button type="button" onClick={() => fields.push({})}>
        Add Member
      </button>
      {(touched || submitFailed) && error && <span>{error}</span>}
    </li>
    {fields.map((member, index) => (
      <li key={index}>
        <button
          type="button"
          title="Remove Member"
          onClick={() => fields.remove(index)}
        />
        <h4>Member #{index + 1}</h4>
        <Field
          name={`${member}.firstName`}
          type="text"
          component={renderField}
          label="First Name"
        />
        <Field
          name={`${member}.lastName`}
          type="text"
          component={renderField}
          label="Last Name"
        />
        <FieldArray name={`${member}.hobbies`} component={renderHobbies} />
      </li>
    ))}
  </ul>
);

let FieldArraysForm = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <form onSubmit={handleSubmit}>
      <Field
        name="clubName"
        type="text"
        component={renderField}
        label="Club Name"
      />
      <FieldArray name="members" component={renderMembers} />
      <div>
        <button type="submit" disabled={submitting}>Submit</button>
        <button type="button" disabled={pristine || submitting} onClick={reset}>
          Clear Values
        </button>
      </div>
    </form>
  );
};

FieldArraysForm = reduxForm({
  form: 'fieldArrays', // a unique identifier for this form
  validate
})(FieldArraysForm)

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
```
It is an example from here https://codesandbox.io/s/8z1r6oyq40
when I press an Add Member button I'll get an error **TypeError: Invalid attempt to spread non-iterable instance**

So the problem in a validation function here is
```javascript
const validate = values => {
  console.log('values = ', values);
  const errors = {};
  if (!values.clubName) {
    errors.clubName = 'Required';
  }
  if (!values.members || !values.members.length) {
    errors.members = { _error: 'At least one member must be entered' };
    console.log('here we are');
  }
...
```
values of the sandbox has
**values =  Object {members: Array[1]}
members: Array[1]**

my values has nothing but **{}**

when I commented this lines
```
if (!values.members || !values.members.length) {
    errors.members = { _error: 'At least one member must be entered' };
    console.log('here we are');
  }
```
the button works,
How to fix this bug? 

package.json
```json
{
  "name": "form",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "npm": "^6.9.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-redux": "^7.0.3",
    "react-scripts": "3.0.0",
    "redux": "^4.0.1",
    "redux-form": "^8.2.0",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0",
    "start": "^5.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

**Comment by Isarstyle** [comment]
+1

**Comment by tomasztomys** [comment]
+1

**Comment by krzysztofgrowinski** [comment]
+1

**Comment by wgottschalk** [comment]
This is a serious issue. The upgrade to redux 4 and react-redux 7 means that version 7 of redux-form is in compatible. We're blocked on the upgrade until the field array issue gets solved.

**Comment by cristian-sima** [comment]
Same here

**Comment by vascojm82** [comment]
Is there anyone looking at any of the new issues on redux-form or is redux-form officially dead as far as development goes?.

**Comment by Asday** [comment]
Worth noting that this is an issue in the _example_ for `FieldArray`.

https://redux-form.com/8.0.4/examples/fieldarrays/ <-- Broken
https://redux-form.com/7.4.2/examples/fieldarrays/ <-- Working

**Comment by Asday** [comment]
https://github.com/erikras/redux-form/issues/4457

**Comment by dmason30** [comment]
duplicate #4439

**Comment by renatoagds** [comment]
Hi all! Made the revert of the PR that caused this problem. Will release a new version in the next few hours. :)

**Comment by Asday** [comment]
That one was for sure the issue, then?

_Siiiick_.  That might also solve #4465 - I'll let you know once the new version's on `npm`!

(Also, your name is _really_ cool).

**Comment by erikras** [comment]
Published fix in [`v8.2.1`](https://github.com/erikras/redux-form/releases/tag/v8.2.1).
